Unix/protocol/protocol.c

```diff
@@ -21,10 +21,68 @@
 #include <pal/file.h>
 #include <pal/sleep.h>
 
-// #define  ENABLE_TRACING 1
+//#define  ENABLE_TRACING 1
 #ifdef ENABLE_TRACING
-# define TRACING_LEVEL 4
-# include <deprecated/logging/logging.h>
+//# define TRACING_LEVEL 4
+#include <base/result.h>
+#include <base/logbase.h>
+#include <base/log.h>
+#define LOGE2 __LOGE
+#define LOGW2 __LOGW
+#define LOGD2 __LOGD
+#define mistrerror Result_ToString
+typedef struct _NameValuePairs
+{
+    unsigned int Value;
+    const char* Name;
+} NameValuePairs;
+static NameValuePairs MessageTagNames[] =
```

Comment:
Since MessageTag is an enum, is it really necessary to do NameValuePairs, and not just something like MessageTagNames?

If you really want Name to pair them, it's probably safer to pair MessageTag (as an enum) with the char* strings.